pysixdesk/lib/preprocess.py

```diff
@@ -1,366 +1,385 @@
 from pysixdesk.lib.pysixdb import SixDB
 from pysixdesk.lib.resultparser import parse_results
 
 
-logger = utils.condor_logger('preprocess')
+class PreprocessJob(SixtrackJob):
+    def __init__(self, task_id, input_info):
+        '''Class to handle the execution of the preprocessing job.
+
+        Args:
+            task_id (int): Current work unit ID.
+            input_info (str/path): Path to the database configuration file.
+
+        Raises:
+            FileNotFoundError: If required input file is not found in database.
+            ValueError: If the provided output files are incorect.
+        '''
+        self._logger = utils.condor_logger('preprocess')
+        self.task_id = task_id
+        cf = configparser.ConfigParser()
+        cf.optionxform = str
+        cf.read(input_info)
+        self.db = SixDB(cf['db_info'].items())
+        db_type = cf['db_info']['db_type']
+        self.db_type = db_type.lower()
+        # cf.clear()
+
+        mask_keys = list(cf['mask'].keys())
+        outputs = self.db.select('preprocess_wu', mask_keys,
+                                 where=f'task_id={self.task_id}')
 
-
-def run(task_id, input_info):
-    cf = configparser.ConfigParser()
-    cf.optionxform = str  # preserve case
-    cf.read(input_info)
-    db_info = dict(cf['db_info'])
-    dbtype = db_info['db_type']
-    db = SixDB(db_info)
-    try:
-        task_id = str(task_id)
-        mask_info = cf['mask']
-        mask_keys = list(mask_info.keys())
-        where = 'task_id=%s' % task_id
-        outputs = db.select('preprocess_wu', mask_keys, where)
         if not outputs[0]:
             content = "Data not found for preprocess task %s!" % task_id
             raise FileNotFoundError(content)
-        db.close()
+
+        self.madx_cfg = cf['madx']
+        self.mask_cfg = dict(zip(mask_keys, outputs[0]))
+        # outputs = self.db.select('preprocess_wu',
+        #                          ['input_file', 'task_id'],
+        #                          f'task_id={task_id}')
+
+        # self.task_id = outputs[0][1]
+        # intput_buf = outputs[0][0]
+        # in_files = utils.decompress_buf(intput_buf, None, 'buf')
+        # cf.read_string(in_files)
+        self.cf = cf
+        # self.madx_cfg = cf['madx']
+
+        output_files = self.madx_cfg["output_files"]
         try:
-            mask_data = dict(zip(mask_keys, outputs[0]))
-            madx_config = cf['madx']
-            mask_config = mask_data
-            oneturn = madx_config['oneturn']
-            collimation = madx_config['collimation']
-            madxjob(madx_config, mask_config)
+            output_files = utils.decode_strings(output_files)
         except Exception:
-            content = 'MADX job failed.'
-            logger.error(content, exc_info=True)
-            if dbtype.lower() == 'sql':
-                return
+            content = "Wrong setting of madx output!"
+            raise ValueError(content)
```

Comment:
I agree that this try-except block (and the other similar ones) is unnecessary for a simple call the to the `decode` function. This try-except block, and the other ones, were already in the original scripts, see lines 201, 268, 274, 139,  of `preprocess.py` in the current master branch for a few examples. I left them in because I didn't want to do too many functional changes when doing major structural changes. I welcome their removal.

Also, as it is, the try-except of the `main` block does not cover errors in the `__init__` method, this is an easy change if you think it should.